data/src/message.rs

```diff
@@ -75,12 +75,29 @@ impl From<Encoded> for proto::Message {
 
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub enum Target {
-    Server { source: Source },
-    Channel { channel: Channel, source: Source },
-    Query { nick: Nick, source: Source },
+    Server {
+        source: Source,
+    },
+    Channel {
+        channel: Channel,
+        source: Source,
+        prefix: Option<char>,
+    },
+    Query {
+        nick: Nick,
+        source: Source,
+    },
 }
 
 impl Target {
+    pub fn prefix(&self) -> &Option<char> {
+        match self {
+            Target::Server { .. } => &None,
+            Target::Channel { prefix, .. } => prefix,
+            Target::Query { .. } => &None,
+        }
+    }
```

Comment:
Nit: Typically you'd return Option<&T>, which you can do by prefix.as_ref()

Reply:
Makes perfect sense!  Thanks for the tip :)